File: finstack-py/tests/golden/runners/pricing_ir_future.py
"""Domain runner for `rates.ir_future` golden fixtures."""

from __future__ import annotations

from tests.golden.schema import GoldenFixture

from .pricing_common import run_pricing_fixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Run one interest-rate future fixture through the Python bindings."""
    return run_pricing_fixture(fixture)

File: finstack/valuations/tests/golden/runners/pricing_ir_future.rs
//! Domain runner for `rates.ir_future` golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Interest-rate future golden runner.
pub struct IrFutureRunner;

impl DomainRunner for IrFutureRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::pricing_common::run_pricing_fixture(fixture)
    }
}
